### ISSUE TITLE
HTTP ERROR 500 : Velocity could not be initialized

### ISSUE THREAD

**Issue by bleachzk** [azkaban/azkaban#917]
azkaban version:3.15.0

java.lang.RuntimeException: JarResourceLoader : JAR path must start with jar: -> see java.net.JarURLConnection for information
	at org.apache.velocity.runtime.resource.loader.JarResourceLoader.loadJar(JarResourceLoader.java:143)
	at org.apache.velocity.runtime.resource.loader.JarResourceLoader.init(JarResourceLoader.java:118)
	at org.apache.velocity.runtime.resource.ResourceManagerImpl.initialize(ResourceManagerImpl.java:143)
	at org.apache.velocity.runtime.RuntimeInstance.initializeResourceManager(RuntimeInstance.java:730)
	at org.apache.velocity.runtime.RuntimeInstance.init(RuntimeInstance.java:263)
	at org.apache.velocity.runtime.RuntimeInstance.requireInitialization(RuntimeInstance.java:302)
	at org.apache.velocity.runtime.RuntimeInstance.getTemplate(RuntimeInstance.java:1531)
	at org.apache.velocity.app.VelocityEngine.mergeTemplate(VelocityEngine.java:343)
	at azkaban.webapp.servlet.Page.render(Page.java:69)
	at azkaban.webapp.servlet.LoginAbstractAzkabanServlet.handleLogin(LoginAbstractAzkabanServlet.java:272)
	at azkaban.webapp.servlet.LoginAbstractAzkabanServlet.handleLogin(LoginAbstractAzkabanServlet.java:262)
	at azkaban.webapp.servlet.LoginAbstractAzkabanServlet.doGet(LoginAbstractAzkabanServlet.java:129)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:668)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:770)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:401)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:766)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:928)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:549)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)

**Comment by kunkun-tang** [comment]
Is the issue still a block? @bleachzk

**Comment by jessica0530** [comment]
how do you fix

**Comment by realman129942** [comment]
how do you fix

**Comment by li-afaris** [comment]
I know this is a late update, but perhaps it will help someone else in the future.  I ran into the `JAR path must start with jar:` message while deploying the Azkaban webserver (version 3.50) into a new environment.  

After looking at the Apache velocity developer guide, I realized `jar.resource.loader.path` had an issue and found both the AzkabanWebServer & TriggerPluginLoad classes dynamically build the jar.resource.load.path.  **If there's an empty directory under 'plugins',** it will trigger the `Jar path must start with jar:` exception.

If one is wondering what to put into the plugins dir, take a look at azkaban-solo-server/build/install/azkaban-solo-server/plugins/ for ideas.
